lirc: plugin ftdi is missing (pkg-config in depsBuildBuild)

### Describe the bug

In nixos-unstable (2022-11-10) the package lirc do not build the ftdi plugin. This plugin is availabe in nixos-22.05.

### Analysis

This issue was introduced with https://github.com/NixOS/nixpkgs/pull/190212, which moved package `pkg-config` from `nativeBuildInputs` to `depsBuildBuild`.
This move renamed the environment variable form `PKG_CONFIG_PATH` to `PKG_CONFIG_PATH_FOR_BUILD` and therefor the program `pkg-config` fails to find package `libftdi1` during configure phase:

```
checking for libftdi1 >= 1.0... no
```

Side note: Do we have a general problem with `pkg-config` in `depsBuildBuild`?

### Expected behavior

The lirc ftdi plugin is available.

### Solving attempts

Moving `pkg-config` from `depsBuildBuild` back to `nativeBuildInputs`, raises following error:

```
python-pkg/lirc/_client.c:1:10: fatal error: Python.h: No such file or directory
    1 | #include <Python.h>
      |          ^~~~~~~~~~
```

The move does not raise an error with lirc version 0.10.2, which is currently available as PR https://github.com/NixOS/nixpkgs/pull/200558.

### Notify maintainers

@SuperSandro2000 , @pSub

--- a/pkgs/development/libraries/lirc/default.nix
+++ b/pkgs/development/libraries/lirc/default.nix
@@ -6,11 +6,11 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "lirc";
-  version = "0.10.1";
+  version = "0.10.2";
 
   src = fetchurl {
     url = "mirror://sourceforge/lirc/${pname}-${version}.tar.bz2";
-    sha256 = "1whlyifvvc7w04ahq07nnk1h18wc8j7c6wnvlb6mszravxh3qxcb";
+    sha256 = "sha256-PUTsgnSIHPJi8WCAVkHwgn/8wgreDYXn5vO5Dg09Iio=";
   };
 
   patches = [

--- a/pkgs/development/libraries/lirc/default.nix
+++ b/pkgs/development/libraries/lirc/default.nix
@@ -50,9 +50,7 @@ stdenv.mkDerivation rec {
 
   strictDeps = true;
 
-  nativeBuildInputs = [ autoreconfHook help2man libxslt pythonEnv ];
-
-  depsBuildBuild = [ pkg-config ];
+  nativeBuildInputs = [ autoreconfHook help2man libxslt pythonEnv pkg-config ];
 
   buildInputs = [ alsa-lib xlibsWrapper systemd libusb-compat-0_1 libftdi1 ];
 

